backend/b2/b2.go

```diff
@@ -1318,16 +1319,22 @@ func (f *Fs) purge(ctx context.Context, dir string, oldOnly bool, deleteHidden b
 				// Check current version of the file
 				if deleteHidden && object.Action == "hide" {
 					fs.Debugf(remote, "Deleting current version (id %q) as it is a hide marker", object.ID)
-					toBeDeleted <- object
+					if !operations.SkipDestructive(ctx, object.Name, "remove hide marker") {
```

Comment:
Yeah - to be consistent with `s3` `purge`, it should only ask you once. For the backend commands, since it's not very obvious exactly what will be deleted, I think it should ask you for each file, since you can always skip the rest of the questions.

For clarity, I think I'll add another argument to the `purge()` function, something like `noSkipDestructive` rather than testing for the combination of existing flags that indicate that it's a `purge` rather than one of the backend operations. Sound good?

Reply:
I should probably have mentioned you in that reply, @ncw 😉